src/app/features/create-event/event-type-form/event-type-form.component.ts

```diff
@@ -0,0 +1,57 @@
+import { ChangeDetectionStrategy, Component, inject, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { EventTypeForm } from '../create-event.interface';
+import { EventFormProvider } from '../event-form-provider';
+
+@Component({
+  selector: 'app-event-type-form',
+  templateUrl: './event-type-form.component.html',
+  styleUrls: ['./event-type-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EventTypeFormComponent implements OnInit {
+  private eventFormProvider = inject(EventFormProvider);
+  private router = inject(Router);
+  private route = inject(ActivatedRoute);
+
+  constructor() {
+    this.eventTypeForm = this.eventFormProvider.getForm().get('eventTypeForm') as FormGroup<EventTypeForm>;
```

Comment:
eventFormProvider powinien być generykiem dzięki czemu unikniemy castowania wartości. Powinniśmy móc to zrobić